starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,31 @@
-function QueueDataStructure () {
+function QueueDataStructure() {
+  this.queueControl = [];
+  this.MAX_SIZE = 100;
+  this.isEmpty = function() {
+    if (this.queueControl.length === 0) {
+      return true;
+    }
+    return false;
+  };
+  this.canEnqueue = function() {
+    if (this.queueControl.length === this.MAX_SIZE) {
+      return false;
+    }
+    return true;
+  };
+  this.enqueue = function(num) {
+    if (this.canEnqueue()) {
+      this.queueControl.unshift(num);
+      return this.queueControl;
+    } else {
+      return "Queue Overflow";
+    }
+  };
+  this.dequeue = function() {
+    if (this.isEmpty === false) {
```

Comment:
This is the reason a test is failing for the queue.
Should be   _if (this.isEmpty() === false)_ , 
or even better so    _if (!this.isEmpty())_
Notice the missing ( ) in your code to invoke the method. Instead you are checking if the **function itself** is false, and this will always return false, since a function object is truthy (javascript quirks).